src/main/java/com/mongodb/jdbc/MongoSQLResultSetMetaData.java

```diff
@@ -50,12 +50,15 @@ private void assertDatasourceSchema(MongoJsonSchema schema) throws SQLException
         }
     }
 
-    private void processDataSource(MongoJsonSchema schema, String datasource) throws SQLException {
+    private void processDataSource(MongoJsonSchema schema, String datasource, boolean sortFieldsAlphabetically) throws SQLException {
         MongoJsonSchema datasourceSchema = schema.properties.get(datasource);
         assertDatasourceSchema(datasourceSchema);
 
         String[] fields = datasourceSchema.properties.keySet().toArray(new String[0]);
-        Arrays.sort(fields);
+        // [SQL-703] Only sort the result columns alphabetically for non database metadata resultsets.
```

Comment:
```suggestion
        // Only sort the result columns alphabetically for non database metadata resultsets.
```
2 things about this comment:
1. If we include it, I'm not sure we should/need to include the ticket number.
2. I'm not sure we need to include here in-line. Perhaps as a method-level comment explaining what the parameter is, and examples of why/when it is used would be better?

Reply:
ah, I think the issue here is that the comment would be better MongoSQLResultset. It shows why there is a flag and what value is used when. I like to flag small changes with big implications with the JIRA id, but I think with the updated comment it is not be necessary to keep it.